### DISCUSSION TITLE
New database integration -- GaussDB

### DISCUSSION THREAD

**Discussion by yanzhifeng-vip** [dbeaver/dbeaver#22087] [DBeaver Development]
Hello, everyone,
DBeaver is an open-source, widely used database management tool with high availability. We hope that GaussDB can be integrated into DBeaver to facilitate the use of GaussDB. GaussDB has many enterprise users. GaussDB is integrated into the DBeaver community, which facilitates users to use and expands the number of DBeaver users. We believe that we will make some contributions to the open source community.

What Is GaussDB?
GaussDB is a distributed relational database from Huawei. It supports intra-city cross-AZ deployment with zero data loss. With a distributed architecture, GaussDB supports petabytes of storage and contains more than 1,000 nodes per DB instance. It is highly available, secure, and scalable and provides services including quick deployment, backup, restoration, monitoring, and alarm reporting for enterprises. Currently, GaussDB has been widely used worldwide. Product official website: https://www.huaweicloud.com/intl/en-us/product/gaussdb.html

Adaptation solution and plan,
Adaptation solution:
Based on the existing modules of MySQL, PostgreSQL, and Generic in the community, try to inherit and rewrite methods to implement the interfaces and functions of the GaussDB plug-in without affecting the parent class.
Adaptation plan:
Initial integration objective: The GaussDB plug-ins ext.gaussdb and ext.gaussdb.ui are added based on the PostgreSQL plug-in. The GaussDB plug-ins can be directly selected for a new connection and the driver can be downloaded. (We will submit a pull request within one month.)
Long-term adaptation objective: Gradually modify the specific functions of GaussDB to further improve the support for GaussDB. For example, you can select different database compatibility modes when creating a database.

Our team is dealing with the community for the first time, and any suggestions or opinions are welcome to us.
@serge-rider @Wroud 

![20231204-190041(WeLinkPC)](https://github.com/dbeaver/dbeaver/assets/65696622/3246fda2-549b-4b11-80ad-a36d634116a7)
![20231204-190026(WeLinkPC)](https://github.com/dbeaver/dbeaver/assets/65696622/22fc1cce-07ab-4717-afc2-1e12ce657006)

**Comment by LonwoLonwo** [comment]
Hello @yanzhifeng-vip 

DBeaver is always open to new contributions.
You can read our article about contributing in DBeaver: https://github.com/dbeaver/dbeaver/wiki/Contribute-your-code

And any questions about coding we can discuss in your PR.

**Reply by yanzhifeng-vip** [reply]
Thank you for your prompt reply. We will submit the PR as soon as possible.
